### PR TITLE
Test auto-update and memory saturation (swap) event

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

File: test/check-application [hunk 1]
```diff
@@ -43,6 +44,27 @@ class TestApplication(testlib.MachineCase):
         # still only one hour
         b.wait_js_func("ph_count_check", ".metrics-hour", 1)
 
+        # should only have at most three valid minutes, the rest should be empty
+        valid_start = b.call_js_func("ph_count", ".metrics-data-cpu.valid-data")
+        self.assertLessEqual(valid_start, 3)
```
Comment (marusak) [inline test/check-application:49]: Why 3 minutes? Is there some reason for this number?

File: test/check-application [hunk 2]
```diff
@@ -43,6 +44,27 @@ class TestApplication(testlib.MachineCase):
...
+        # there are often some initial events at machine startup
+        initial_events = b.call_js_func("ph_count", ".metrics-events")
+
+        # memory saturation event
+        mem_hog = m.spawn("MEMBLOB=$(yes | dd bs=1M count=500 iflag=fullblock); sleep infinity", "mem_hog.log")
+        time.sleep(30)
+        m.execute("kill %d" % mem_hog)
+        # UI auto-updates every minute, so it should pick up the new spike within the next 30s
+        b.wait_js_func("(exp => ph_count('.metrics-events') == exp)", initial_events + 1)
+        # in our CI this ends up as a net swap event, and no actual used memory increase
```
Comment (marusak) [inline test/check-application:65]: Good start for sure :)
We would like to also check that the svg actually is some kind of spike-like shape or something, right?

File: test/check-application (hunk 1 above)
Comment (martinpitt) [reply test/check-application:49]: Just to be on the safe side for utterly slow CI, and sometimes our VM images run into late 90s boot delays due to some failing services.

File: test/check-application (hunk 2 above)
Comment (martinpitt) [reply test/check-application:65]: Hmm, we could check that at least one X coordinate is > 0.8 perhaps? Testing graph shapes seems a bit finnicky. If you have good ideas, go wild of course!

Comment (marusak) [reply test/check-application:65]: yeah, I was thinking of having some function that would take some coefficient and would check that the spike is higher than than `check_spike(selector_for_the_spike, 0.8)` or so. The function would collect the x-coordinates of the given shape and check it exceeds it. But that is surely good for next iterations :)

**Pull request merged**